pilot/cmd/pilot-agent/status/server.go

```diff
@@ -110,6 +114,13 @@ func NewServer(config Config) (*Server, error) {
 	if err := json.Unmarshal([]byte(config.KubeAppProbers), &s.appKubeProbers); err != nil {
 		return nil, fmt.Errorf("failed to decode app prober err = %v, json string = %v", err, config.KubeAppProbers)
 	}
+	// setup a prometheus stats exporter for istio-agent.
+	exporter, err := ocprom.NewExporter(ocprom.Options{Registry: prometheus.DefaultRegisterer.(*prometheus.Registry)})
+	if err != nil {
+		return nil, fmt.Errorf("could not set up prometheus exporter: %v", err)
+	}
+	view.RegisterExporter(exporter)
```

Comment:
@douglas-reid I feel like there must be a way to get this info without actually listening on a real server, but haven't found one. Any ideas?

Reply:
agree, but is it all so bad to have an endpoint dedicated to istio-agent's health?

Reply:
I'm a bit confused... Having an exporter seems right. You want to have something translating to prom from OC.

But, isn't the agent already doing aggregation of the envoy and application stats?  Can't we just overload that handler to combine with that dump instead of a new listerning port?

Just have the handler that is handling `/stats/prometheus` call  `exporter.ServeHTTP` and then do what is already doing?

Reply:
@douglas-reid not sure how that would work. How would I record the response?